geode-management/src/main/java/org/apache/geode/management/configuration/RegionShortcut.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.configuration;
+
+import org.apache.geode.cache.configuration.RegionType;
+
+/**
+ * these are the region shortcuts supported by Cluster Management V2 API. The attributes of these
+ * region shortcuts are the same as their namesakes in RegionShortcut
+ */
+public enum RegionShortcut {
+  PARTITION,
+  PARTITION_REDUNDANT,
+  PARTITION_PERSISTENT,
+  PARTITION_REDUNDANT_PERSISTENT,
+  PARTITION_OVERFLOW,
+  PARTITION_REDUNDANT_OVERFLOW,
+  PARTITION_PERSISTENT_OVERFLOW,
+  PARTITION_REDUNDANT_PERSISTENT_OVERFLOW,
+  PARTITION_HEAP_LRU,
+  PARTITION_REDUNDANT_HEAP_LRU,
+
+  PARTITION_PROXY,
+  PARTITION_PROXY_REDUNDANT,
+
+  REPLICATE,
+  REPLICATE_PERSISTENT,
+  REPLICATE_OVERFLOW,
+  REPLICATE_PERSISTENT_OVERFLOW,
+  REPLICATE_HEAP_LRU,
+
+  REPLICATE_PROXY;
+
+  /**
+   * @return the corresponding region type of each of these region shortcut
+   */
+  public RegionType getRegionType() {
+    if (withPartition()) {
```

Comment:
Instead of implementing this with a bunch of logic, just add a "private final RegionType regionType" field to this enum class. You can initialize each corresponding regionType when you declare the enums and then this method can just return this.regionType.

Reply:
I tried that, but this looks cleaner. The other way makes the shortcut enum class cluttered, but it's just my opinion.